src/solver/simulation/economy.cpp

```diff
@@ -146,7 +146,53 @@ bool Economy::year(Progression::Task& progression,
 
         try
         {
-            OPT_OptimisationHebdomadaire(pProblemesHebdo[numSpace], numSpace);
+            
+            if (pProblemesHebdo[numSpace]->UseAdequacyPatch)
+            {
+                logs.debug() << "Use adequacy patch: 1st run...";
+                //todo need to change interconnection links property before 1st run
+                pProblemesHebdo[numSpace]->AdequacyFirstStep = true;
+                OPT_OptimisationHebdomadaire(pProblemesHebdo[numSpace], numSpace);
+                pProblemesHebdo[numSpace]->AdequacyFirstStep = false;
+            
+                int numberOfAreas = pProblemesHebdo[numSpace]->NombreDePays;
+                int numberOfTimesteps = pProblemesHebdo[numSpace]->NombreDePasDeTemps;
+
+                //save DENS from 1st run
+                for(int pays = 0; pays < numberOfAreas; ++pays)
+                {
+                    for(int step = 0; step < numberOfTimesteps; ++step)
+                    {
+                        pProblemesHebdo[numSpace]->ResultatsHoraires[pays]->ValeursHorairesDENS[step]
+                            = pProblemesHebdo[numSpace]->ResultatsHoraires[pays]->ValeursHorairesDeDefaillancePositive[step];
+                        // logs.debug() << "##0:" << pays << ":" << step << ":" << pProblemesHebdo[numSpace]->ResultatsHoraires[pays]->ValeursHorairesDENS[step];
+                    }
+                }            
+
+                //todo need to change interconnection link property before second run
+                ::SIM_RenseignementProblemeHebdo(*pProblemesHebdo[numSpace], state, numSpace, hourInTheYear); //??? it is correct
```

Comment:
What does this line do ? 
The rest of the code is clear to me. 
I'll try to test it.

Reply:
reinit/reset something; we are asking rte; you can have a look if you find something